### PR TITLE
More Improvements using Location

### PR DESCRIPTION
**Pull request by spencer-lunarg**
some misc improvements for use of `Location`

### CODE REVIEW

File: layers/core_checks/cc_shader_interface.cpp [hunk 1]
```diff
@@ -101,10 +101,8 @@ bool CoreChecks::ValidateInterfaceVertexInput(const PIPELINE_STATE &pipeline, co
                                           "] Vertex attribute at location %" PRIu32 " not consumed by vertex shader",
                                           pipeline.create_index, location);
         } else if (!attribute_input && shader_input) {
-            skip |= LogError(module_state.handle(), "VUID-VkGraphicsPipelineCreateInfo-Input-07904",
-                             "vkCreateGraphicsPipelines(): pCreateInfos[%" PRIu32
-                             "] Vertex shader consumes input at location %" PRIu32 " but not provided",
-                             pipeline.create_index, location);
+            skip |= LogError("VUID-VkGraphicsPipelineCreateInfo-Input-07904", module_state.handle(), create_info_loc,
+                             "Vertex shader consumes input at location %" PRIu32 " but not provided.", location);
```
Comment (arno-lunarg) [inline layers/core_checks/cc_shader_interface.cpp:105]:
```suggestion
                             "specifies a vertex shader that consumes unprovided input at location %" PRIu32 ".", location);
```

**Review by arno-lunarg — APPROVED** [review]
LGTM!
Not sure about the suggestions in `cc_shader_interface.cpp`

File: layers/core_checks/cc_cmd_buffer.cpp [hunk 2]
```diff
@@ -45,37 +45,17 @@ bool CoreChecks::ReportInvalidCommandBuffer(const CMD_BUFFER_STATE &cb_state, co
...
     bool skip = false;
     for (uint32_t i = 0; i < commandBufferCount; i++) {
         auto cb_state = GetRead<CMD_BUFFER_STATE>(pCommandBuffers[i]);
         // Delete CB information structure, and remove from commandBufferMap
-        if (cb_state) {
-            skip |= CheckCommandBufferInFlight(cb_state.get(), "free", "VUID-vkFreeCommandBuffers-pCommandBuffers-00047");
+        if (cb_state && cb_state->InUse()) {
+            const LogObjectList objlist(pCommandBuffers[i], commandPool);
+            skip |= LogError("VUID-vkFreeCommandBuffers-pCommandBuffers-00047", objlist,
+                             error_obj.location.dot(Field::pCommandBuffers, i), "Attempt to free %s which is in use.",
```
[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply layers/core_checks/cc_cmd_buffer.cpp:57]: agree, I guess I just moved what we had into this message, but now there is no needing to pass `"free"` into an argument, agree your change is nicer (and cleaners)

File: layers/core_checks/cc_shader_interface.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/core_checks/cc_shader_interface.cpp:105]: this and the other ones here just need more information passed in ... going to just redo these

**Pull request merged**